src/main.rs

```diff
@@ -380,11 +434,90 @@ fn main() {
 					},
 				}
 			}
-			println!("Building for {}...", MinSec::since(starttime));
-			if workers.iter().all(|worker| *worker == WorkerStatus::Done) {
+
+			let build_is_done = buildstate.all_workers_done();
+
+			// Compute remaining time for this build, by simulating a build.
+			let mut simulated_time = Instant::now();
+			let mut estimation_impossible = false;
+			while !buildstate.all_workers_done_or_idle() {
+				// Give all simulated workers something to do
+				for i in 0..buildstate.workers.len() {
+					let worker = &buildstate.workers[i];
+					match worker {
+						WorkerStatus::Starting | WorkerStatus::Idle => {
+							let next = queuestate.next_at(simulated_time);
+							buildstate.set_status(i, match next {
+								Some(task) => WorkerStatus::Running{task},
+								None => WorkerStatus::Idle,
+							});
+						}
+						_ => {}
+					}
+				}
+
+				// Find the job with the lowest remaining time
+				let earliest_remaining_job = match buildstate.workers.iter()
+					.enumerate()
+					.flat_map(|(i, worker)| {
+						match worker {
+							WorkerStatus::Running{task, ..} => {
+								let taskstatus = queuestate.get_task_status(*task);
+								match taskstatus {
+									TaskStatus::Running{start_time, ..} => {
+										let runtime = if simulated_time > start_time { simulated_time - start_time } else { Duration::from_millis(0) };
+										match estimated_total_task_time_func(*task) {
+											Some(time) => Some((i, task, time.checked_sub(runtime).unwrap_or(Duration::from_millis(0)))),
+											None => None,
+										}
+									},
+									TaskStatus::Finished{..} => Some((i, task, Duration::from_millis(0))),
+									_ => unreachable!(),
+								}
+							},
+							_ => None,
+						}
+					})
+					.min_by_key(|&(_, _, time)| time)
+				{
+					Some(earliest_remaining_job) => earliest_remaining_job,
+					None => {
+						estimation_impossible = true;
+						break;
+					}
+				};
+
+				// Pass that time
+				simulated_time += earliest_remaining_job.2;
+
+				// Complete that task, if it isn't already Finished
+				match queuestate.get_task_status(*earliest_remaining_job.1) {
+					TaskStatus::Running{ .. } => {queuestate.complete_task_at(*earliest_remaining_job.1, true, simulated_time);},
```

Comment:
You can omit the `{` and `;}`.

Reply:
That lead to an error, because `complete_task_at` returned something, while the other `match` case didn't...